### PR TITLE
5 2 report panel headings

### PR DESCRIPTION
**Pull request by alex40724**

### CODE REVIEW

File: src/UI/templates/default/Panel/tpl.report.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="panel panel-primary">
+<div class="panel panel-primary il-panel-report">
```
Comment (Amstutz) [inline src/UI/templates/default/Panel/tpl.report.html:1]: Thanks for adding il-panel-report. IMHO, this is exactly how we should use CSS classes. il-[name-of-family]-[name-of-component] on top level and all CSS followed from there. Nice.

**Comment by alex40724** [comment]
I am unsure how we want to provide a consistent heading structure in the new UI framework. The current `panel > sub panel > card structure` resulted in a `h3 > h3 > h2` hierarchy which is clearly a bug to me. I changed it to `h3 > h4 > h5`. I think for the panels this is ok in general. The cards now use a h5, which might result in missing heading levels in other contexts, e.g. `h3 > h5`. But I think this is less problematic than using the same or even higher heading numbers for substructures.

**Comment by alex40724** [comment]
See also http://www.ilias.de/mantis/view.php?id=19420

**Review by Amstutz — APPROVED** [review]
I checked the Deck of Cards that I have in use with the proposed changes. Looks good to me.

Regarding headings: Do we have a strict rule, not to leave out headings? If not, would you recommend to introduce one? If so, we might think of not using hX in components like cards, that could be used in potentially different levels of hierarchy. Wie could e.g. just use il-card-title or similar.

**Review by Amstutz — CHANGES_REQUESTED** [review]
Attention the Unit tests are currently failing due to the DOM changes made. Please fix.

I fixed them locally, but can not push it to you repo due to permission restrictions. You can merge them from here: https://github.com/Amstutz/ILIAS/tree/5_2_report_panel_headings

**Comment by alex40724** [comment]
@Amstutz Thanks for the test fixes.

**Pull request merged**